container_registry/container_analysis/samples.py

```diff
@@ -0,0 +1,343 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from time import sleep
+
+from google.api_core.exceptions import AlreadyExists
+from google.cloud.devtools.containeranalysis_v1alpha1 \
+    import container_analysis_client
+from google.cloud.devtools.containeranalysis_v1alpha1.proto \
+    import containeranalysis_pb2
+from google.cloud.devtools.containeranalysis_v1alpha1.proto \
+    import package_vulnerability_pb2
+from google.cloud.pubsub import SubscriberClient
+
+
+# [START create_note]
+def create_note(note_id, project_id):
+    """Creates and returns a new note
+
+    :param note_id: A user-specified identifier for the note.
+    :param project_id: the GCP project the note will be created under
+    :return: the newly created note
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    parent = client.project_path(project_id)
+
+    type = package_vulnerability_pb2.VulnerabilityType()
+    note = containeranalysis_pb2.Note(vulnerability_type=type)
+    response = client.create_note(parent, note_id, note)
+    return response
+# [END create_note]
+
+
+# [START create_occurrence]
+def create_occurrence(image_url, parent_note_id, project_id):
+    """Creates and returns a new occurrence
+
+    :param image_url: the Container Registry URL associated with the image
+                example: "https://gcr.io/project/image@sha256:foo"
+    :param parent_note_id: the identifier of the associated note
+    :param project_id: the GCP project the occurrence will be created under
+    :return: the newly created occurrence
+    :raises
+        google.api_core.exceptions.GoogleAPICallError: If the request
+            failed for any reason.
+        google.api_core.exceptions.RetryError: If the request failed due
+            to a retryable error and retry attempts failed.
+        ValueError: If the parameters are invalid.
+    """
+    client = container_analysis_client.ContainerAnalysisClient()
+    parent_name = client.note_path(project_id, parent_note_id)
+    project_name = client.project_path(project_id)
+    vul = package_vulnerability_pb2.VulnerabilityType.VulnerabilityDetails()
+
+    occurrence = containeranalysis_pb2.Occurrence(note_name=parent_name,
```

Comment:
Prefer to avoid hanging indents:

    occurence = containeranalysis_pb2.Occurence(
            note_name=parent_name,
            resource_url=image_url,
            vulnerability_details=vul)